proto/redpanda/api/dataplane/v1alpha1/kafka_connect.proto

```diff
@@ -0,0 +1,328 @@
+// connector
+enum HolisticState {
+  HOLISTIC_STATE_UNSPECIFIED = 0;
+  // PAUSED: The connector/task has been administratively paused.
+  HOLISTIC_STATE_PAUSED = 1;
+  // RESTARTING: he connector/task is restarting.
+  HOLISTIC_STATE_RESTARTING = 2;
+  // DESTROYED: Connector is in destroyed state, regardless of any tasks.
+  HOLISTIC_STATE_DESTROYED = 3;
+  // STOPPED: The connector/task has been stopped.
+  HOLISTIC_STATE_STOPPED = 4;
+  // The connector/task has not yet been assigned to a worker
+  // UNASSIGNED: Connector is in unassigned state.
+  //    Or Connector is in running state, and there are unassigned tasks.
+  HOLISTIC_STATE_UNASSIGNED = 5;
+  // HEALTHY: Connector is in running state, > 0 tasks, all of them in running state.
+  HOLISTIC_STATE_HEALTHY = 6;
+  // UNHEALTHY: Connector is failed state.
+  //			Or Connector is in running state but has 0 tasks.
+  // 			Or Connector is in running state, has > 0 tasks, and all tasks are in failed state.
+  HOLISTIC_STATE_UNHEALTHY = 7;
+  // DEGRADED: Connector is in running state, has > 0 tasks, but has at least one state in failed state, but not all tasks are failed.
+  HOLISTIC_STATE_DEGRADED = 8;
+  // UNKNOWN: The connector/task could no be determined
+  HOLISTIC_STATE_UNKNOWN = 9;
+}
+
+message ConnectorPlugin {
+  string type = 1;
+  string version = 2;
+  string class = 3;
+}
+
+message ConnectCluster {
+  message Info {
+    string version = 1;
+    string commit = 2;
+    string kafka_cluster_id = 3;
+  }
+  string name = 1;
+  string address = 2;
+  Info info = 3;
+  repeated ConnectorPlugin plugins = 4;
+}
+
+message ConnectorStatus {
+  message Connector {
+    string state = 1;
+    string worker_id = 2;
+  }
+  string name = 1;
+  Connector connector = 2;
+  repeated TaskStatus tasks = 3;
+  string type = 4;
+  string trace = 5;
+}
+
+message TaskStatus {
+  int32 id = 1;
+  string state = 2;
+  string worker_id = 3;
+  string trace = 4;
+}
+
+message TaskInfo {
+  string connector = 1;
+  int32 task = 2;
+}
+
+// ConnectorError is the error of a connector, this is holistic error
+// abstraction, made parsing the error trace of connector or Task
+message ConnectorError {
+  enum Type {
+    TYPE_UNSPECIFIED = 0;
+    TYPE_ERROR = 1;
+    TYPE_WARNING = 2;
+  }
+  Type type = 1;
+  string title = 2;
+  string content = 3;
+}
+
+// ConectorInfo is the spec of the connector, as defined in the Kafka connect
+// API, it can be used as input of the connector creation or output of the
+// connectors
+message ConnectorSpec {
+  string name = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true
+  ];
+
+  map<string, string> config = 2 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true
+  ];
+  repeated TaskInfo tasks = 3 [(google.api.field_behavior) = OUTPUT_ONLY];
+  string type = 4 [(google.api.field_behavior) = OUTPUT_ONLY];
+}
+
+message ListConnectorsRequest {
+  repeated string expand = 1 [
```

Comment:
What do you think about removing the expand parameter and always setting both info and status to true? Personally I don't find this very idiomatic, but at least we have a stable response schema now, so that I'm fine with keeping it as well.

To me it feels like they just added these as optional parameters at a later point to remain 100% backwards compatible (no additional fields added in newer versions).

Reply:
That makes sense removing it